pkg/services/searchV2/auth.go

```diff
@@ -19,14 +20,62 @@ type simpleSQLAuthService struct {
 	sql *sqlstore.SQLStore
 }
 
+type dashIdQueryResult struct {
+	UID string `xorm:"uid"`
+}
+
 func (a *simpleSQLAuthService) GetDashboardReadFilter(user *models.SignedInUser) (ResourceFilter, error) {
-	if user == nil || user.HasRole(models.ROLE_ADMIN) {
-		return alwaysTrueFilter, nil
+	// this filter works on the legacy `dashboard_acl` table
+	// we will also need to use `accesscontrol` after https://github.com/grafana/grafana/pull/44702/files is merged
+	// see https://github.com/grafana/grafana/blob/e355bd6d3a04111b8c9959f85e81beabbeb746bf/pkg/services/sqlstore/permissions/dashboard.go#L84
+	filter := permissions.DashboardPermissionFilter{
+		OrgRole:         user.OrgRole,
+		OrgId:           user.OrgId,
+		Dialect:         a.sql.Dialect,
+		UserId:          user.UserId,
+		PermissionLevel: models.PERMISSION_VIEW,
+	}
+
+	rows := make([]*dashIdQueryResult, 0)
+
+	err := a.sql.WithDbSession(context.Background(), func(sess *sqlstore.DBSession) error {
+		sql, params := filter.Where()
+		sess.Table("dashboard").
+			Where(sql, params...).
+			Cols("uid")
+
+		err := sess.Find(&rows)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	var uids []string
+	for i := 0; i < len(rows); i++ {
+		uids = append(uids, rows[i].UID)
+	}
+
+	return uidFilter{uids: uids}.filter, err
```

Comment:
```suggestion
	return &uidFilter{uids: uids}.filter, err
```
May resolve lint issue - but better to re-check

Reply:
transformed `resourceFilter` into an interface to fix this

Reply:
Sorry - this won't work, need to define a separate variable. Maybe if ResourceFilter was an interface things were more readable.